libraries/Microsoft.Bot.Builder.LanguageGeneration/Templates.cs

```diff
@@ -391,15 +394,53 @@ private void CheckErrors()
             }
         }
 
-        private bool GetStrictModeFromOptions(IList<string> options)
+        private EvaluationOptions GetCurrentEvalOptions()
+        {
+            var opt = GetEvaluationOptions(Options);
+            foreach (var templates in References) 
+            {
+                var childOpt = GetEvaluationOptions(templates.Options);
+                if (childOpt.StrictMode != null)
+                {
+                    opt.StrictMode = childOpt.StrictMode;
+                }
+
+                if (childOpt.NullSubstitution != null)
+                {
+                    opt.NullSubstitution = childOpt.NullSubstitution;
+                }
+
+                if (childOpt.LineBreakStyle != null)
+                {
+                    opt.LineBreakStyle = childOpt.LineBreakStyle;
+                }
+            }
+
+            //normalize option to non-null value for evaluation
+            if (opt.StrictMode == null)
+            {
+                opt.StrictMode = false;
+            }
+
+            if (opt.LineBreakStyle == null)
+            {
+                opt.LineBreakStyle = LGLineBreakStyle.Default;
+            }
+
+            return opt;
+        }
+
+        private EvaluationOptions GetEvaluationOptions(IList<string> options)
```

Comment:
we really should extract this function to EvaluationOptions class, and add a similiar function `ApplyOptionFromLines(param string[] lines)`

Reply:
Now it is static method in EvaluationOptions class